build_scripts/build_usd.py

```diff
@@ -1574,9 +1619,22 @@ def InstallUSD(context, force, buildArgs):
 group.add_argument("--build", type=str,
                    help=("Build directory for USD and 3rd-party dependencies " 
                          "(default: <install_dir>/build)"))
+
+group.add_argument("--build-debug", dest="build_debug", action="store_true",
+                    help="Build in Debug mode")
+
+group.add_argument("--build-release", dest="build_release", action="store_true",
+                    help="Build in Release mode")
+
+group.add_argument("--build-relwithdebug", dest="build_relwithdebug", action="store_true",
+                    help="Build in RelWithDebInfo mode")
+
 group.add_argument("--build-args", type=str, nargs="*", default=[],
                    help=("Custom arguments to pass to build system when "
                          "building libraries (see docs above)"))
+group.add_argument("--build-python-info", type=str, nargs=4, default=[],
+                   metavar=('PYTHON_EXECUTABLE', 'PYTHON_INCLUDE_DIR', 'PYTHON_LIBRARY', 'PYTHON_VERSION'),
+                   help=("Specify a custom python to use during build"))
```

Comment:
Flag to specify a custom python. We use this at Autodesk to pass in the Python from Maya, which from latest Maya can be either py2 or py3. I've seen other requests in usd interest to build with a custom python.

Reply:
No unfortunately not. Maya 2022 is dual-python containing both python 2.7 and 3.7 and mayapy/mayapy2. The internal function IsMayaPython() no longer works with those and since it was marked with comments about not having any special Maya knowledge I decided not to modify it. But rather create a more generic ability to specify the python to build with via a new flag `--build-python-info`. 

This allows a user to call the build_usd.py with any python (such as system) but pass in the python that you want USD to use to build the python bindings with.

I even recall a thread (usd-interest/github?) where someone asked for something like this. Unfortunately I cannot find it.

In MayaUsd repo we are using `find_package(Python 3.7)` and also have the same mechanism to set the python to use (via build flag).

Reply:
OK, understood. The only thing I'd ask then is if we could put a bit of documentation about this in the program description section around lines 1559 or so. A brief blurb with exactly what you said about letting users specify the Python for USD to use and a list of the four parameters to pass to `--build-python-info` would be perfect.